cmake/RippledCompiler.cmake

```diff
@@ -91,7 +91,7 @@ if (MSVC)
       -machine:X64)
 else ()
   # HACK : because these need to come first, before any warning demotion
-  string (APPEND CMAKE_CXX_FLAGS " -Wall -Wdeprecated")
+  string (APPEND CMAKE_CXX_FLAGS " -Wall -Wno-deprecated-declarations")
```

Comment:
* Is there any way we can rewrite `Expected` to avoid this warning?
* Is there any way we can localize the workaround to `Expected.h`?

Reply:
Both good points
1. That's work in progress, in a different repo
2. Yes ! But we also need workaround in `hash_append.h` (which is not bad, considering this is `clang` tripping on `libstdc++` headers)